### ISSUE TITLE
Attachment error responses are not machine parsable

### ISSUE THREAD

**Issue by sjregan** [buddypress/BP-REST#315]
When uploading attachments via the API there are several errors that are returned that contain human readable information but do not provide enough metadata so that they can be programmatically handled.

eg.

```
return new WP_Error(
	"bp_rest_attachments_{$this->object}_cover_upload_error",
	__( 'The BuddyPress attachments uploads directory is not set.', 'buddypress' ),
	array(
		'status' => 500,
	)
);
```

```
return new WP_Error(
	"bp_rest_attachments_{$this->object}_cover_upload_error",
	sprintf(
		/* translators: %$1s and %$2s is replaced with the correct sizes. */
		__( 'You have selected an image that is smaller than recommended. For better results, make sure to upload an image that is larger than %1$spx wide, and %2$spx tall.', 'buddypress' ),
		(int) $cover_dimensions['width'],
		(int) $cover_dimensions['height']
	),
	array(
		'status' => 500,
	)
);
```

I believe additional data should be returned that is not translated:

```
return new WP_Error(
	"bp_rest_attachments_{$this->object}_cover_upload_error",
	__( 'The BuddyPress attachments uploads directory is not set.', 'buddypress' ),
	array(
		'status' => 500,
                'reason' => 'upload_dir_not_set',
	)
);
```

or

```
return new WP_Error(
	"bp_rest_attachments_{$this->object}_cover_upload_error",
	sprintf(
		/* translators: %$1s and %$2s is replaced with the correct sizes. */
		__( 'You have selected an image that is smaller than recommended. For better results, make sure to upload an image that is larger than %1$spx wide, and %2$spx tall.', 'buddypress' ),
		(int) $cover_dimensions['width'],
		(int) $cover_dimensions['height']
	),
	array(
		'status' => 500,
                'reason' => 'image_too_small',
                'min_width' => (int) $cover_dimensions['width'],
                'min_height' => (int) $cover_dimensions['height']
	)
);
```

Any thoughts?

**Comment by renatonascalves** [comment]
I wonder what's the use case for this?

@imath What are you thoughts about this?

**Comment by sjregan** [comment]
@renatonascalves For me it is to display feedback to the user about the error, however it is not always appropriate to use the translated string in its entirety (eg. UX considerations, theme/app design)

Also the error message is written as though the upload has been successful and the message is just additional helpful information regarding the successful upload - so we need to display a different message.

**Comment by renatonascalves** [comment]
I'm inclined to say this should be handled in your app/implementation. The current implementation 
aligns with the BuddyPress one as it returns an error object, status and reason of the error.

However you decide to translate or update that to your audience/app, is up to you. Otherwise we run into the issue of use cases and specific situations.

**Comment by sjregan** [comment]
Thanks for the feedback, I appreciate it.

I am trying to handle it in my app, the problem is programmatically speaking the app cannot tell the difference between the error types, they all have status 500. 

Without returning additional information with the error codes, the app cannot know if it is a misconfiguration on the server causing the error, or the user uploading the wrong image.

Just displaying the error message results in a poor experience as the end user has no idea what they should do differently when they are shown 'The BuddyPress attachments uploads directory is not set.'.

**Comment by imath** [comment]
@renatonascalves I like the idea, if it needs to be implemented within the app, then we should probably include add filters to error response with the needed data as paremeters.

**Comment by ExoGeek** [comment]
Totally agree. I ran into the same problem...

**Comment by imath** [comment]
@sjregan or @ExoGeek could you add a PR to this issue with the filters and values you need ?

**Comment by renatonascalves** [comment]
Closing this isse since the code was merged at #323
